packages/cli/src/utils/sessionCleanup.ts

```
@@ -0,0 +1,277 @@
+    // Verify file exists before attempting deletion
+    await fs.access(sessionPath, fs.constants.F_OK);
+
+    // Attempt to read and validate the session file structure
+    // This ensures we're not deleting corrupted or non-session files
+    const content = await fs.readFile(sessionPath, 'utf8');
+    const sessionData = JSON.parse(content);
+
+    // Basic validation that this is actually a session file
+    if (
+      !sessionData.sessionId ||
+      !sessionData.messages ||
+      !Array.isArray(sessionData.messages)
+    ) {
+      throw new Error('Invalid session file structure');
+    }
+
+    // Perform the deletion
+    await fs.unlink(sessionPath);
+
+    if (debugMode) {
+      console.debug(
+        `Deleted expired session: ${session.id} (${session.lastUpdated})`,
+      );
+    }
+  } catch (error) {
+    if (error instanceof Error) {
+      throw new Error(
+        `Failed to delete session ${session.id}: ${error.message}`,
+      );
+    }
+    throw error;
+  }
+}
+
+/**
+ * Parses retention period strings like "30d", "7d", "24h" into milliseconds
+ */
+function parseRetentionPeriod(period: string): number {
+  const match = period.match(/^(\d+)([dhwm])$/);
+  if (!match) {
+    return 0; // Invalid format
+  }
+
+  const value = parseInt(match[1], 10);
+  const unit = match[2];
+
+  const multipliers = {
+    h: 60 * 60 * 1000, // hours to ms
+    d: 24 * 60 * 60 * 1000, // days to ms
+    w: 7 * 24 * 60 * 60 * 1000, // weeks to ms
+    m: 30 * 24 * 60 * 60 * 1000, // months (30 days) to ms
+  };
+
+  return value * multipliers[unit as keyof typeof multipliers] || 0;
+}
+
+/**
+ * Validates retention configuration
+ */
+function validateRetentionConfig(config: SessionRetentionSettings): {
+  valid: boolean;
+  error?: string;
+} {
+  if (!config.enabled) {
+    return { valid: false, error: 'Retention not enabled' };
+  }
+
+  // Validate maxAge if provided
+  if (config.maxAge) {
+    const maxAgeMs = parseRetentionPeriod(config.maxAge);
```

Please change parseRetentionPeriod to raise an error if the format is invalid and handle the error here.